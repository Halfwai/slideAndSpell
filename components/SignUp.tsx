import { Animated, TouchableOpacity, StyleSheet, View, TextInput, Alert, Modal, ScrollView, Linking, Pressable } from "react-native"
import MyAppText from "@/components/MyAppText"
import { useEffect, useRef, useState } from "react"

import AuthButton from "@/components/AuthButton"

import { supabase } from '@/lib/supabase'

import { COLOURS } from '@/constants/colours'
import PrivacyPolicy from "@/components/PrivacyPolicy"
interface SignUpProps {
    setMenu: Function,
}

export default function SignUp(props: SignUpProps) {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [passwordConfirm, setPasswordConfirm] = useState('')
    const [displayName, setDisplayName] = useState('')
    const [loading, setLoading] = useState(false)

    const [showPrivacyPolicy, setShowPrivacyPolicy] = useState(false)


    async function signUpWithEmail() {
        if (email === '') {
            Alert.alert('Please enter an email');
            return;
        };
        if (password === '') {
            Alert.alert('Please enter a password');
            return;
        }
        if (password !== passwordConfirm) {
            Alert.alert('Passwords do not match');
            return;
        }
        if (displayName === '') {
            Alert.alert('Please enter a display name');
            return;
        }
        setLoading(true)
        const {
            data: { session },
            error,
        } = await supabase.auth.signUp({
            email: email,
            password: password,
            options: {
                data: {
                    display_name: 'username'
                }
            }
        })

        if (error) Alert.alert(error.message)
        setLoading(false)
    }

    return (
        <View style={styles.container}>
            <MyAppText style={styles.title}>Sign Up</MyAppText>
            <View style={styles.inputContainer}>
                <TextInput
                    placeholder="Email"
                    style={styles.input}
                    value={email}
                    onChangeText={setEmail}
                    autoComplete="email"
                    autoCorrect={false}
                    inputMode="email"
                />
                <TextInput
                    placeholder="Display Name"
                    style={styles.input}
                    value={displayName}
                    onChangeText={setDisplayName}
                    autoCorrect={false}
                />
                <TextInput
                    placeholder="Password"
                    style={styles.input}
                    value={password}
                    onChangeText={setPassword}
                    autoCorrect={false}
                    secureTextEntry={true}
                />
                <TextInput
                    placeholder="Confirm Password"
                    style={styles.input}
                    value={passwordConfirm}
                    onChangeText={setPasswordConfirm}
                    autoCorrect={false}
                    secureTextEntry={true}
                />
            </View>
            <TouchableOpacity 
                style={styles.privacyPolicy}
                onPress={() => {
                    setShowPrivacyPolicy(true)
                }
            }>
                <MyAppText>
                    By Signing up you agree to our <MyAppText
                        style={styles.privacyPolicyText}
                    >Privacy Policy</MyAppText>
                </MyAppText>
            </TouchableOpacity>
            <AuthButton text="Sign Up" onPress={signUpWithEmail} style={{ backgroundColor: COLOURS.green }} />
            <AuthButton text="Back" onPress={() => { props.setMenu("welcome") }} style={{ backgroundColor: "white", borderColor: COLOURS.green }} />
            <Modal
                visible={showPrivacyPolicy}
                animationType="slide"
                transparent={true}
            >
                <PrivacyPolicy
                    hidePrivacyPolicy={() => setShowPrivacyPolicy(false)}
                />

            </Modal>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
    },
    inputContainer: {
        width: '100%',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
    },
    privacyPolicy: {
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
    },
    privacyPolicyText: {
        textDecorationLine: 'underline',
    },
    input: {
        width: '90%',
        marginTop: 10,
        borderWidth: 1,
        borderRadius: 10,
        backgroundColor: 'white',
        padding: 10,
        fontSize: 20,
        height: 50,
    },
})