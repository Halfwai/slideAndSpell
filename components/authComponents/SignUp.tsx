import { TouchableOpacity, StyleSheet, View, TextInput, Modal } from "react-native"
import MyAppText from "@/components/common/MyAppText"
import React, { useState } from "react"
import AuthButton from "@/components/buttons/AuthButton"
import { signUpWithEmail } from '@/utils/supabaseFunctions'
import { COLOURS } from '@/constants/colours'
import PrivacyPolicy from "@/components/modals/PrivacyPolicy"

interface SignUpProps {
    setMenu: Function,
}

export default function SignUp(props: SignUpProps) {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [passwordConfirm, setPasswordConfirm] = useState('')
    const [displayName, setDisplayName] = useState('')
    const [loading, setLoading] = useState(false)
    const [showPrivacyPolicy, setShowPrivacyPolicy] = useState(false)

    return (
        <View style={styles.container}>
            <MyAppText style={styles.title}>Sign Up</MyAppText>
            <View style={styles.inputContainer}>
                <TextInput
                    placeholder="Email"
                    style={styles.input}
                    value={email}
                    onChangeText={setEmail}
                    autoComplete="email"
                    autoCorrect={false}
                    inputMode="email"
                />
                <TextInput
                    placeholder="Display Name"
                    style={styles.input}
                    value={displayName}
                    onChangeText={setDisplayName}
                    autoCorrect={false}
                />
                <TextInput
                    placeholder="Password"
                    style={styles.input}
                    value={password}
                    onChangeText={setPassword}
                    autoCorrect={false}
                    secureTextEntry={true}
                />
                <TextInput
                    placeholder="Confirm Password"
                    style={styles.input}
                    value={passwordConfirm}
                    onChangeText={setPasswordConfirm}
                    autoCorrect={false}
                    secureTextEntry={true}
                />
            </View>
            <TouchableOpacity
                style={styles.privacyPolicy}
                onPress={() => {
                    setShowPrivacyPolicy(true)
                }
                }>
                <MyAppText>
                    By Signing up you agree to our <MyAppText
                        style={styles.privacyPolicyText}
                    >Privacy Policy</MyAppText>
                </MyAppText>
            </TouchableOpacity>
            <AuthButton
                text="Sign Up"
                onPress={async() => {
                    setLoading(true);
                    await signUpWithEmail(email, password, passwordConfirm, displayName);
                    setLoading(false);
                }}
                style={{ backgroundColor: COLOURS.green }}
                disabled={loading}
            />
            <AuthButton
                text="Back"
                onPress={() => {
                    props.setMenu("welcome")
                }}
                style={{ backgroundColor: "white", borderColor: COLOURS.green }} 
                disabled={loading}
            />                
            <Modal
                visible={showPrivacyPolicy}
                animationType="slide"
                transparent={true}
            >
                <PrivacyPolicy
                    hidePrivacyPolicy={() => setShowPrivacyPolicy(false)}
                />

            </Modal>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
    },
    inputContainer: {
        width: '100%',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
    },
    privacyPolicy: {
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
    },
    privacyPolicyText: {
        textDecorationLine: 'underline',
    },
    input: {
        width: '90%',
        marginTop: 10,
        borderWidth: 1,
        borderRadius: 10,
        backgroundColor: 'white',
        padding: 10,
        fontSize: 20,
        height: 50,
    },
})