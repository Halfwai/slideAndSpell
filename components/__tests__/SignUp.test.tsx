import { act, render } from '@testing-library/react-native';
import SignUp from '@/components/authComponents/SignUp';

jest.mock('@/lib/supabase', () => ({
    supabase: {

    },
}));

it(`SignUp Component renders correctly`, async() => {
    const { toJSON } = render(
        <SignUp
            setMenu={() => {}}
        />
    );
    expect(toJSON()).toMatchSnapshot();
});