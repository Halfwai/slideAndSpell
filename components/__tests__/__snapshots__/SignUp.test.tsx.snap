// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Quicksand",
        },
        {
          "fontSize": 30,
        },
      ]
    }
  >
    Sign Up
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "width": "100%",
      }
    }
  >
    <TextInput
      autoComplete="email"
      autoCorrect={false}
      inputMode="email"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "marginTop": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Display Name"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "marginTop": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "marginTop": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Confirm Password"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "marginTop": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          undefined,
        ]
      }
    >
      By Signing up you agree to our 
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Privacy Policy
      </Text>
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#8BF0BC",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginBottom": 20,
        "opacity": 1,
        "padding": 12,
        "width": "90%",
      }
    }
    testID="authButton"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#8BF0BC",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginBottom": 20,
        "opacity": 1,
        "padding": 12,
        "width": "90%",
      }
    }
    testID="authButton"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Back
    </Text>
  </View>
</View>
`;
